### ISSUE TITLE
Configurable Idle Time Settings

### ISSUE THREAD

**Issue by kingster** [relayrides/pushy#452]
In our production environment, our corporate firewall has much lesser idle timeout settings(causing connection drops, which can be avoided by setting correct idle_time), and currently pushy has the following settings as hard coded values

```java
static final int PING_IDLE_TIME_MILLIS = 60_000; // milliseconds
```

Does it make sense to set this via the ApnsClientBuilder?

**Comment by jchambers** [comment]
Sure. Why not?

**Comment by kingster** [comment]
Just had one question, In `ApnsClientHandler` there is this

```java
private static final int PING_TIMEOUT = 30; // seconds
```

As far as I understand this is the PING reply timeout, and there is an assert 

```java
assert PING_TIMEOUT < ApnsClient.PING_IDLE_TIME_MILLIS;
```
Right now PING_TIMEOUT has been set as half of idle time, should this be the scenario always?

**Comment by kingster** [comment]
Ah, that assert is also buggy! It's comparing seconds with milliseconds! :(

**Comment by jchambers** [comment]
> Right now PING_TIMEOUT has been set as half of idle time, should this be the scenario always?

`PING_TIMEOUT` must be less than `PING_IDLE_TIME`; we want to make sure that the "hey, the ping timed out" task appears in the event queue before the "fire the next `PING` frame" task. There are many roads we can take to get there. A mathematical relationship between the two seems reasonable if we're allowing callers to specify `PING_IDLE_TIME`; letting callers specify both seems needlessly complicated and error-prone.

> Ah, that assert is also buggy! It's comparing seconds with milliseconds!

Yes. This is already fixed in another branch.

**Comment by jchambers** [comment]
Oops—we merged #453 a few days ago, which closes this issue. Thanks again for the contribution!
